### PR TITLE
PR for 2.0.0

### PR DESCRIPTION
**Pull request by apeltzer**
## PR checklist for 2.0.0 (opening this on behalf of the smrnaseq team 🥳 

Mostly so that we can all have a final round of review to get things done asap. 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [x] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](<https://github.com/>nf-core/smrnaseq/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/smrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -76,42 +77,38 @@
...
-                    "default": "ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.gz"
-                },
-                "bt_index": {
-                    "type": "string",
-                    "description": "Path to the Bowtie 1 reference genome index.",
-                    "fa_icon": "fas fa-indent"
+                    "default": "https://mirbase.org/ftp/CURRENT/hairpin.fa.gz"
                 },
-                "references_parsed": {
+                "bt_indices": {
```
Comment (JoseEspinosa) [inline nextflow_schema.json:97]: Would it be more self-explanatory just `bowtie_index`?

Comment (JoseEspinosa) [reply nextflow_schema.json:97]: If yes, all the instances where the parameter is used should be changed.

File: subworkflows/local/mirdeep2.nf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+workflow MIRDEEP2 {
+    take:
+    reads        // channel: [ val(meta), [ reads ] ]
+    fasta
+    indices
+    hairpin
+    mature
+
+    main:
+
```
Comment (JoseEspinosa) [inline subworkflows/local/mirdeep2.nf:18]:
```suggestion
```

File: subworkflows/local/genome_quant.nf [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+include { BOWTIE_MAP_SEQ as BOWTIE_MAP_GENOME } from '../../modules/local/bowtie_map_mirna'
+
+workflow GENOME_QUANT {
+    take:
+    fasta
+    bt_index
+    reads      // channel: [ val(meta), [ reads ] ]
+
+    main:
+
```
Comment (JoseEspinosa) [inline subworkflows/local/genome_quant.nf:16]:
```suggestion
```

File: subworkflows/local/mirna_quant.nf [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+workflow MIRNA_QUANT {
+    take:
+    mature     // channel: fasta file
+    hairpin    // channel: fasta file
+    gtf        // channle: GTF file
+    reads      // channel: [ val(meta), [ reads ] ]
+
+    main:
+
```
Comment (JoseEspinosa) [inline subworkflows/local/mirna_quant.nf:34]:
```suggestion
```

File: subworkflows/local/mirtrace.nf [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+//
+
+include { MIRTRACE_RUN } from '../../modules/local/mirtrace'
+
+workflow MIRTRACE {
+    take:
+    reads      // channel: [ val(meta), [ reads ] ]
+
+    main:
+
```
Comment (JoseEspinosa) [inline subworkflows/local/mirtrace.nf:12]:
```suggestion
```

File: subworkflows/nf-core/bam_sort_samtools.nf [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+include { SAMTOOLS_INDEX     } from '../../modules/nf-core/modules/samtools/index/main'
+include { BAM_STATS_SAMTOOLS } from './bam_stats_samtools'
+
+workflow BAM_SORT_SAMTOOLS {
+    take:
+    ch_bam // channel: [ val(meta), [ bam ] ]
+    fasta
+
+    main:
+
```
Comment (JoseEspinosa) [inline subworkflows/nf-core/bam_sort_samtools.nf:15]:
```suggestion
```

File: subworkflows/nf-core/bam_stats_samtools.nf [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+include { SAMTOOLS_IDXSTATS } from '../../modules/nf-core/modules/samtools/idxstats/main'
+include { SAMTOOLS_FLAGSTAT } from '../../modules/nf-core/modules/samtools/flagstat/main'
+
+workflow BAM_STATS_SAMTOOLS {
+    take:
+    ch_bam_bai // channel: [ val(meta), [ bam ], [bai/csi] ]
+    fasta
+
+    main:
+
```
Comment (JoseEspinosa) [inline subworkflows/nf-core/bam_stats_samtools.nf:15]:
```suggestion
```

File: subworkflows/nf-core/fastqc_trimgalore.nf [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+include { TRIMGALORE       } from '../../modules/local/trimgalore'
+
+workflow FASTQC_TRIMGALORE {
+    take:
+    reads         // channel: [ val(meta), [ reads ] ]
+    skip_fastqc   // boolean: true/false
+    skip_trimming // boolean: true/false
+
+    main:
+
```
Comment (JoseEspinosa) [inline subworkflows/nf-core/fastqc_trimgalore.nf:15]:
```suggestion
```

File: workflows/smrnaseq.nf [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+    .set { ch_cat_fastq }
+    ch_versions = ch_versions.mix(CAT_FASTQ.out.versions.first().ifEmpty(null))
+
+    //
+    // SUBWORKFLOW: mirtrace QC
+    //
+    MIRTRACE (ch_cat_fastq)
+    ch_versions = ch_versions.mix(MIRTRACE.out.versions.ifEmpty(null))
+
+
```
Comment (JoseEspinosa) [inline workflows/smrnaseq.nf:132]:
```suggestion
```

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Awesome job! 🚀 
Mainly small suggestion.
I also realized that local modules are not using `ext.args` to pass the arguments might be worth implementing it so that users can change the parameters given to a tool, but this might also be implemented on a new release along with `mirgenedb`

File: modules/local/mirdeep2_mapper.nf [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    $reads \\
+    -e \\
+    -h \\
+    -i \\
+    -j \\
+    -m \\
+    -p $index_base \\
+    -s ${meta.id}_collapsed.fa \\
+    -t ${meta.id}_reads_vs_refdb.arf \\
+    -o 4
```
[the remark this answers is not in the document]
Comment (apeltzer) [reply modules/local/mirdeep2_mapper.nf:34]: Hmm fair point!

**Comment by apeltzer** [comment]
Agree with @JoseEspinosa that we could/should probably add the `ext.args` possibility toallow for more flexibility but this could be done in the next release (2.1.0) also integrating mirgenedb + contamination filtering possibilities in one go. Regarding renaming of the bt_index channel, I agree this could be a favourable thing to have so I'm going to change that to be a bit more explicit.

**Review by JoseEspinosa — APPROVED** [review]
Thanks @apeltzer LGTM! 😄
Created this small one to fix the linting: https://github.com/nf-core/smrnaseq/pull/157

**Review by sguizard — APPROVED** [review]
Apart no default value for `--outdir`, LGMT!

**Pull request merged**